htslib/sam.h update breaks compatibility with gcc 4.4.7

With one of the latest updates I cannot compile HTSlib with gcc 4.4.7 anymore. The error is the following:
```
$ gcc -Wall -g -O2 -I.  -c -o hts.o hts.c
In file included from ./cram/cram_samtools.h:61,
                 from cram/cram.h:45,
                 from hts.c:41:
./htslib/sam.h:63: error: redefinition of typedef ‘bam_hdr_t’
htslib/hts.h:209: note: previous declaration of ‘bam_hdr_t’ was here
```
The problem seems to be that in `htslib/hts.h` you have one definition of `bam_hdr_t`:
```
typedef struct bam_hdr_t bam_hdr_t;
```
While in `htslib/sam.h` you have a different definition of `bam_hdr_t`:
```
typedef struct bam_hdr_t {
    int32_t n_targets, ignore_sam_err;
    uint32_t l_text;
    uint32_t *target_len;
    int8_t *cigar_tab;
    char **target_name;
    char *text;
    void *sdict;
    uint32_t ref_count;
} bam_hdr_t;
```
I don't really understand the reason why older version of gcc don't see this as an issue. Is it possible to restructure the code to restore compatibility with gcc 4.4.7?

--- a/htslib/hts.h
+++ b/htslib/hts.h
@@ -42,6 +42,7 @@ extern "C" {
 typedef struct BGZF BGZF;
 #define HTS_BGZF_TYPEDEF
 #endif
+struct bam_hdr_t;
 struct cram_fd;
 struct hFILE;
 struct hts_tpool;
@@ -206,8 +207,6 @@ typedef struct htsFormat {
 struct __hts_idx_t;
 typedef struct __hts_idx_t hts_idx_t;
 
-typedef struct bam_hdr_t bam_hdr_t;
-
 // Maintainers note htsFile cannot be an opaque structure because some of its
 // fields are part of libhts.so's ABI (hence these fields must not be moved):
 //  - fp is used in the public sam_itr_next()/etc macros
@@ -228,7 +227,7 @@ typedef struct {
     htsFormat format;
     hts_idx_t *idx;
     const char *fnidx;
-    bam_hdr_t *bam_header;
+    struct bam_hdr_t *bam_header;
 } htsFile;
 
 // A combined thread pool and queue allocation size.

